### DISCUSSION TITLE
Chess Plugin Does Not Support Variants

### DISCUSSION THREAD

**Discussion by Ombucha** [lowlighter/metrics#1407] [Q&A]
Since some variants allow moves which are not allowed in standard chess, they seem to cause errors while rendering the animation.

**Comment by lowlighter** [comment]
Hi 👋 !

It's possible, the parsing is not done directly by this app but by [chess.js](https://github.com/jhlywa/chess.js/) using the [PGN](https://en.wikipedia.org/wiki/Portable_Game_Notation) format:
https://github.com/lowlighter/metrics/blob/2ca1f471e1a7a920183a649910efd4bd18efeb86/source/plugins/community/chess/index.mjs#L35-L37

But the render is only done by extracting start and end position so it may be a bit wonky if it's not send back correctly or the resulting move has not the same structure.
It has been previously reported that En passant was not drawn properly either, so it might be related

**Reply by Ombucha** [reply]
Can you not use Lichess' built-in feature?

> https://lichess1.org/game/export/gif/{color}/{gameId}.gif?theme={theme}&piece={piece}

Replace the bracketed values with appropriate values for each game to generate a GIF.
This also fixes the issue you mentioned (#1340).

**Reply by lowlighter** [reply]
I didn't know about this endpoint tbh 😅 
It isn't documented in their [API](https://lichess.org/api#tag/Games/operation/gamePgn) so I don't know if it's officially supported and available through token authentication, where you did find the reference for this endpoint ?

**Reply by Ombucha** [reply]
This is not an officially documented endpoint, but can still be utilised. It can be found on any Lichess game page (in the `Share and Export` section).

Token authentication is not required for this endpoint.
Note that you would still need to use [the current endpoint](https://lichess.org/api#tag/Games/operation/apiGamesUser) to fetch the value of `gameId`.
